src/Avalonia.Controls/SizeChangedEventArgs.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Avalonia.Interactivity;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Provides data specific to a SizeChanged event.
+    /// </summary>
+    public class SizeChangedEventArgs : RoutedEventArgs
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        public SizeChangedEventArgs(RoutedEvent? routedEvent)
+            : base (routedEvent)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        /// <param name="source">The source object that raised the routed event.</param>
+        public SizeChangedEventArgs(RoutedEvent? routedEvent, IInteractive? source)
+            : base(routedEvent, source)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        /// <param name="source">The source object that raised the routed event.</param>
+        /// <param name="previousSize">The previous size (or bounds) of the object.</param>
+        /// <param name="newSize">The new size (or bounds) of the object.</param>
+        public SizeChangedEventArgs(
+            RoutedEvent? routedEvent,
+            IInteractive? source,
+            Size previousSize,
+            Size newSize)
```

Comment:
Parameter ordering in other places seems to follow (old, new) so that is what is done here. For example, AvaloniaPropertyChangedEventArgs itself uses this ordering.